python/lsst/ts/pmd/component.py

```diff
@@ -186,23 +187,30 @@ async def get_slots_position(self):
                 split_reply = reply.decode().split(":")
                 position[i] = float(split_reply[-1])
             else:
-                position[i] = math.nan
-                # Try to recover multiplexer
-                self.log.error(
-                    f"Sensor {name} is not reading. {self.read_error_count} continuous read errors reported"
-                )
-
-                if self.read_error_count == 2 and self.multiplexer_recovery():
-                    self.log.info("Multiplexer recovered")
-                    self.read_error_count = 0
-                else:
-                    self.read_error_count += 1
+                while not self.read_error_count >= 3 and not self.mutliplexer_recovered:
```

Comment:
I think that you do not need self.mutliplexer_recovered and also that you do not need read_error_count to be an instance variable. Consider something like this:
```
read_error_count = 0
for i, name in enumerate(self.names):
    if reply ~= b"\r":
        split_reply = reply.decode().split(":")
        position[i] = float(split_reply[-1])
    else:
        while read_error_count < 3:
            #... try to recover
            if await self.multiplexer_recovery():
                self.log.info(...)
                break
        else:
            read_error_count += 1
            self.log.error(...recovery failed...)
```